### ISSUE TITLE
Question about KFIOUloss

### ISSUE THREAD

**Issue by lalalagogogochong** [open-mmlab/mmrotate#216]
你好，请问一下kfiouloss函数中的这两个参数是什么含义呢，没太理解这两个参数的作用。pred_decode，targets_decode
![image](https://user-images.githubusercontent.com/44855506/163147902-053bdeb0-75ce-4efd-bdfc-03ba308d377a.png)

**Comment by yangxue0827** [comment]
The pred box and target box after decoding.

**Comment by lalalagogogochong** [comment]
> decoding

decoding的作用是什么呢，直接对pred box and target box计算kfiou可以吗

**Comment by zytx121** [comment]
pred box and target box 是偏移量，需要 decode 之后才是通常意义上的 box。

**Comment by yangxue0827** [comment]
pred/target is offset, pred_decode/target_decode is box after decoding, kfiou needs box to be calculated.

**Comment by lalalagogogochong** [comment]
> pred/target is offset, pred_decode/target_decode is box after decoding, kfiou needs box to be calculated.

如果我在别的项目中使用kfiouloss，我的pred/target 就是box,是不需要解码的吧，这个解码也是KFIOU组成的一部分吗，我看您的论文解说中好像没有提这个解码，谢谢你了

**Comment by yangxue0827** [comment]
`KFIoU loss = Lc  + Lkf`.   
`Lc` needs offset (pred and target) to claculate center pont `(x,y)` loss .
`Lkf` needs box (pred_decode and target_decode) to be calculated.

**Comment by lalalagogogochong** [comment]
> `KFIoU loss = Lc + Lkf`. `Lc` needs offset (pred and target) to claculate center pont `(x,y)` loss . `Lkf` needs box (pred_decode and target_decode) to be calculated.

谢谢你啦，还有个小问题想问一下，为什么求center pont (x,y) loss 时要用两个box中心点的偏移量计算呢，直接对两个中心点的坐标做一个差不可以吗

**Comment by yangxue0827** [comment]
we refer to the smooth l1. Directly computing the Euclidean distance of two points like GWD may make the loss function lose scale invariance.

**Comment by 78Jarre** [comment]
Hi, Sir.Where is the relevant code for calculating the offset of the center point coordinates?

**Comment by 78Jarre** [comment]
> > `KFIoU loss = Lc + Lkf`.  needs offset (pred and target) to claculate center pont  loss .  needs box (pred_decode and target_decode) to be calculated.`Lc``(x,y)``Lkf`
> 
> 谢谢你啦，还有个小问题想问一下，为什么求center pont (x,y) loss 时要用两个box中心点的偏移量计算呢，直接对两个中心点的坐标做一个差不可以吗

你好，请问pred,target,pred_decode和target_decode这四个参数传递都是什么呢?

**Comment by huahua454545** [comment]
> pred/target 是 offset，pred_decode/target_decode 是解码后的 box，kfiou 需要计算 box。

Why does the target bbox have offset? I don't understand it very well

**Comment by huahua454545** [comment]
> > `KFIoU loss = Lc + Lkf`. 需要偏移 （PRED 和 TARGET） 来固定中心 Pont 损失。 需要计算 Box （pred_decode 和 target_decode）。`Lc``(x,y)``Lkf`
> 
> 谢谢你啦，还有个小问题想问一下，为什么求center pont （x，y） loss 时要用两个box中心点的偏移量计算呢，直接对两个中心点的坐标做一个差不可以吗

Hello, may I ask how to find the offset of target?
